roles/cert_verify/tasks/check_endpoint_serial.yml

```diff
@@ -0,0 +1,35 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Get public endpoint cert serial
+  vars:
+    cert_verify_endpoint: "{{ cert_verify_endpoint_cert_serial.URL.split('/').2 }}"
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command: "{{ lookup('env', 'HOME') }}/bin/oc exec -n openstack -t openstackclient -- bash -c \"echo 'QUIT'|openssl s_client -showcerts -connect {{ cert_verify_endpoint }}:443 2>&1|openssl x509 -noout -serial\""
```

Comment:
ditto regarding line length and the possibility to not pipe in openssl, but output in a file and use the x509_certificate_info module.

Also, we usually push the `when` condition right after the `name:` - it makes reading/filtering slightly easier.